### PR TITLE
TSQL: Improve function coverage

### PR DESCRIPTION
**Pull request by jimidle**
Improves TSQL coverage by adding missing functions to the relevant tables and sharing common functions between dialects.

### CODE REVIEW

File: core/src/test/scala/com/databricks/labs/remorph/parsers/common/FunctionBuilderSpec.scala [hunk 1]
```diff
@@ -476,11 +494,31 @@ class FunctionBuilderSpec extends AnyFlatSpec with Matchers with TableDrivenProp
   "buildFunction" should "Apply known TSql conversion strategies" in {
     val functionBuilder = new TSqlFunctionBuilder
 
-    val result1 = functionBuilder.buildFunction("ISNULL", Seq(ir.Column("x"), ir.Literal(integer = Some(0))))
-    result1 match {
+    var result = functionBuilder.buildFunction("ISNULL", Seq(ir.Column("x"), ir.Literal(integer = Some(0))))
```
Comment (vil1) [inline core/src/test/scala/com/databricks/labs/remorph/parsers/common/FunctionBuilderSpec.scala:497]: Please avoid using `var`.

A more practical approach would be to define a table property check:
```scala
val functions = Table(
  ("inFunctionName", "outFunctionName"),
  ("GET_BIT", "GETBIT"),
  // etc.
)
forAll(functions){(inFn, outFn) =>
  functionBuilder.buildFunction(inFn) match {
    case f:ir.CallFunction => f.function_name shouldBe outFn
    case _ => fail(s"$inFn TSql conversion failed")
  }
}
```

Also, there's a typo on line 520 (><)'

**Review by vil1 — CHANGES_REQUESTED** [review]
please see my comment about `var`

File: core/src/test/scala/com/databricks/labs/remorph/parsers/common/FunctionBuilderSpec.scala (hunk 1 above)
Comment (jimidle) [reply core/src/test/scala/com/databricks/labs/remorph/parsers/common/FunctionBuilderSpec.scala:497]: OK - yeah I guess that is better. In tests I am usually just looking to get it going but example is not implemented in that Spec I think

Comment (jimidle) [reply core/src/test/scala/com/databricks/labs/remorph/parsers/common/FunctionBuilderSpec.scala]: Change - please approve changes.

**Review by vil1 — APPROVED** [review]
lgtm

**Pull request merged**